op-batcher/batcher/channel_builder.go

```diff
@@ -186,7 +192,7 @@ func (c *ChannelBuilder) AddBlock(block *types.Block) (*derive.L1BlockInfo, erro
 		return l1info, fmt.Errorf("adding block to channel out: %w", err)
 	}
 
-	c.blocks = append(c.blocks, block)
+	c.blocks.Enqueue(block)
```

Comment:
Another question - why are the `blocks` & `frames` queues not protected from concurrent access (i understand this is the way the code was before as well)? I see that `op-service/queue` does not wrap append/read operations with an `RWMutex` either.

Reply:
You are correct that `op-service/queue` is not designed to be inherently concurrency safe. The concurrency control is managed by the surrounding code where necessary.

The only async code in use at the moment (i.e. go routines) is in the `driver`, which protects the entire state (including all `Channels` and their `ChannelBuilders`) with a mutex. So I don't have concerns about concurrency, although it might be a good idea to run the race detector on some tests to see if that finds anything.

Reply:
Good to close if this is not a concern. Just stood out to be, hence asked.